src/main.ts

```diff
@@ -1,10 +1,61 @@
-import { App, Construct, Stack, Fn } from '@aws-cdk/core';
+import { App, Construct, Stack, StackProps, Fn } from '@aws-cdk/core';
 import * as eks from '@aws-cdk/aws-eks';
 import * as ec2 from '@aws-cdk/aws-ec2';
 import * as iam from '@aws-cdk/aws-iam';
 
-export class Demo extends Construct {
-  constructor(scope: Construct, id: string) {
+/**
+ * Props for VpcPeering
+ */
+export interface VpcPeeringProps {
+  /**
+   * The existing VPC
+   */
+  readonly vpc: ec2.IVpc;
+  /**
+   * The new peer VPC
+   */
+  readonly peerVpc: ec2.IVpc;
+}
+
+/**
+ * Create VPC peering connection from `vpc` to `peerVpc`
+ */
+export class VpcPeering extends Construct {
+  readonly peeringConnection: ec2.CfnVPCPeeringConnection
+  constructor(scope: Construct, id: string, props: VpcPeeringProps) {
+    super(scope, id)
+
+    const peeringConnection = new ec2.CfnVPCPeeringConnection(this, 'VpcPeeringConnection', {
+      vpcId: props.vpc.vpcId,
+      peerVpcId: props.peerVpc.vpcId
+    })
+    this.peeringConnection = peeringConnection
+
+    // existingVpc(private subnets) --> peering --> newVpc
+    new ec2.CfnRoute(this, 'Existing2NewRoute', {
+      routeTableId: props.vpc.privateSubnets[0].routeTable.routeTableId,
+      destinationCidrBlock: props.vpc.vpcCidrBlock,
+      vpcPeeringConnectionId: peeringConnection.ref,
+    })
+    // newVpc(private subnets) --> peering --> existingVpc
+    new ec2.CfnRoute(this, 'New2ExistingRoute', {
+      routeTableId: props.peerVpc.privateSubnets[0].routeTable.routeTableId,
+      destinationCidrBlock: props.vpc.vpcCidrBlock,
+      vpcPeeringConnectionId: peeringConnection.ref,
+    })
```

Comment:
create two routes for private subnets via `VpcPeeringConnection` to the other Vpc